libsolidity/ast/AST.h

```diff
@@ -928,13 +942,14 @@ class EventDefinition: public CallableDeclaration, public Documented
 
 /**
  * Pseudo AST node that is used as declaration for "this", "msg", "tx", "block" and the global
- * functions when such an identifier is encountered. Will never have a valid location in the source code.
+ * functions when such an identifier is encountered. Will never have a valid location in the source code
+ * and m_id will be set to zero for all instances.
  */
 class MagicVariableDeclaration: public Declaration
 {
 public:
-	MagicVariableDeclaration(ASTString const& _name, Type const* _type):
-		Declaration(SourceLocation(), std::make_shared<ASTString>(_name)), m_type(_type) {}
+	MagicVariableDeclaration(ASTString const& _name, Type const* _type, int _id):
```

Comment:
I think the id can always be derived from the name - shouldn't we call the translation function from the constructor here?